### PR TITLE
 NAMESPACE in own column: kubectl --all-namespaces

### PR DESCRIPTION
**Pull request by erictune**
For `kubectl --all-namespaces get <kind>`, put namespace in own NAMESPACE column rather than using "namespace/name" in NAME column.

See #10626 for discussion.

However, does not close that issue.

Also:
    Report an error if someone asks for --all-namespaces
    when getting a thing that is not namespaced.

```
This is in preparation for a subsequent commit which prints namespace
as its own column.

Restructured test to expect an error for non-namespaced things.
Dropped the part where it was trying to test that not printing
namespace didn't contain namespace.  Some other test can cover that.
```

### CODE REVIEW

**Comment by erictune** [comment]
GCE failure appears to be due to commit becoming unmergeable.

**Comment by erictune** [comment]
e2e failure is:
```
Summarizing 1 Failure:

[Fail] Probing container [It] with readiness probe should not be ready before initial delay and never restart 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/container_probe.go:54
```

**Comment by erictune** [comment]
That test doesn't seem to use kubectl, so it must be a flake.
@quinton-hoole @ixdy are we documenting jenkins e2e flakes somewhere?

**Comment by ixdy** [comment]
This particular flake is #10622.

**Comment by quinton-hoole** [comment]
In general 
here: https://github.com/GoogleCloudPlatform/kubernetes/issues/9312
... but it's overdue for an update - I'm busy with that now.

**Comment by derekwaynecarr** [comment]
I prefer NAMESPACE as a separate column, so big +1 from me on this change
versus the previous mangling of the NAME column.

**Comment by derekwaynecarr** [comment]
I thought PersistentVolume is not namespaced and only PersistentVolumeClaim
is namespaced.

**Comment by quinton-hoole** [comment]
@erictune In general flaky tests are documented in #9312 and https://github.com/GoogleCloudPlatform/kubernetes/blob/0b2efd8d287ab3893cdcd33f4470eaf9ad96ecd4/hack/jenkins/e2e.sh#L72

The former required updating, which I'm doing now.

**Comment by derekwaynecarr** [comment]
Events are namespaced

**Comment by erictune** [comment]
@derekwaynecarr thanks for the review.  duh.  fixed.

**Comment by jlowdermilk** [comment]
LGTM

Request: There are 2 Examples in [cmd_test.go](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/cmd_test.go#L244) that test full output of a human-readable printer. Update one to print using withNamespace=true?

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -723,12 +738,10 @@ func printNodeList(list *api.NodeList, w io.Writer, withNamespace bool, wide boo
 }
 
 func printPersistentVolume(pv *api.PersistentVolume, w io.Writer, withNamespace bool, wide bool, columnLabels []string) error {
-	var name string
 	if withNamespace {
-		name = types.NamespacedName{pv.Namespace, pv.Name}.String()
-	} else {
-		name = pv.Name
+		return fmt.Errorf("persustentVolume is not namespaced")
```
Comment (derekwaynecarr) [inline pkg/kubectl/resource_printer.go]: Typo

**Comment by erictune** [comment]
@derekwaynecarr fixed typo

@jlowdermilk modified example.

**Comment by erictune** [comment]
@bgrant0607 to affirm 1.0-worthyness

**Comment by bgrant0607** [comment]
LGTM
ok to merge
if the errors for non-namespaced objects turn out to be a problem, we can remove them later

**Pull request merged**